### ISSUE TITLE
Readiness endpoint

### ISSUE THREAD

**Issue by lynshi** [microsoft/CCF#5476]
**Is your feature request related to a problem? Please describe.**
Azure Load Balancers provide [TCP, HTTP, and HTTPS health probes](https://learn.microsoft.com/en-us/azure/load-balancer/load-balancer-custom-probe-overview) for determining when a backend pool instance should be allowed to receive requests via the load balancer.

In CCF, a running node is not necessarily ready to receive requests. For example, a newly started node may not be trusted in the network yet so it should not be accessible via the load balancer.

**Describe the solution you'd like**
CCF should provide an endpoint that indicates the readiness of the node to receive requests for the CCF network. A list of conditions (possibly not exhaustive) include:
- Node is trusted.
- Node has finished deserializing the private ledger.
- Node has not received a stop notice.

**Comment by achamayou** [comment]
@lynshi would that be a GET /node/ready, and would it 200 normally and 503 (?) otherwise, with an empty response body?

**Comment by achamayou** [comment]
There is another choice on how to expose these endpoints, we could either put there directly on the relevant frontends:

  - `/app/ready`
  - `/gov/ready`

That's nice from an implementation perspective (the readiness takes care of itself, and the caller must be robust to rejected connections in any case), but pollutes the app namespace somewhat.

Or put them on the node, and include logic to check if the frontends are open:

  - `/node/ready/app` (or some variation)
  - `/node/ready/gov`

This requires cross-frontend logic, but pollutes the app namespace left.

The main difference between app and gov, other than the separate opening times, are the different node states that are acceptable. In states "partOfPublicNetwork" and "readingPrivateLedger", governance is possible.

So to summarise, I would be in favour of split endpoints, under "/node". I am indifferent to the exact naming, path vs query parameter and specific return codes used. @lynshi please confirm your preferences.

**Comment by lynshi** [comment]
According to [Kubernetes docs](https://kubernetes.io/docs/tasks/configure-pod-container/configure-liveness-readiness-startup-probes/#define-a-liveness-http-request) anything `>= 200, < 400` is success and Azure services like [Container Apps](https://learn.microsoft.com/en-us/azure/container-apps/health-probes?tabs=arm-template#http-probes) follow the same convention, so we should be good with `204` to be more semantically correct as there is no body. `503` should also be a reasonable error code.

As discussed on the call today, `/node/ready/app` and `/node/ready/gov` are preferred to avoid polluting the app namespace.
